### PR TITLE
added my-widget-service for Python

### PR DESCRIPTION
**Pull request by rhboyd**
*Issue #, if available:*

*Description of changes: Created a version of the MyWidgetService that uses API Gateway Service Integrations to directly interact with S3.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: python/my-widget-service/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
+# Welcome to your CDK Python project!
+
+This is a blank project for Python development with CDK.
```
Comment (tom98784) [inline python/my-widget-service/README.md]: This file isn't normally included with the example apps on this repo. If it is included, it needs updating to not still have the default text, e.g. 'This is a blank project...'

File: python/my-widget-service/my_widget_service/my_widget_service.egg-info/PKG-INFO [hunk 2]
```diff
@@ -0,0 +1,80 @@
+Metadata-Version: 2.1
+Name: my-widget-service
+Version: 0.0.1
+Summary: An empty CDK Python app
```
Comment (tom98784) [inline python/my-widget-service/my_widget_service/my_widget_service.egg-info/PKG-INFO]: .egg-info files don't need including

File: python/my-widget-service/my_widget_service/my_widget_service.egg-info/SOURCES.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+README.md
```
Comment (tom98784) [inline python/my-widget-service/my_widget_service/my_widget_service.egg-info/SOURCES.txt]: .egg-info files don't need including

File: python/my-widget-service/my_widget_service/my_widget_service.egg-info/dependency_links.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
Comment (tom98784) [inline python/my-widget-service/my_widget_service/my_widget_service.egg-info/dependency_links.txt]: .egg-info files don't need including

File: python/my-widget-service/my_widget_service/my_widget_service.egg-info/requires.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+aws-cdk.core
```
Comment (tom98784) [inline python/my-widget-service/my_widget_service/my_widget_service.egg-info/requires.txt]: .egg-info files don't need including

File: python/my-widget-service/my_widget_service/my_widget_service.egg-info/top_level.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
Comment (tom98784) [inline python/my-widget-service/my_widget_service/my_widget_service.egg-info/top_level.txt]: .egg-info files don't need including

File: python/my-widget-service/my_widget_service/my_widget_service_stack.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
+from aws_cdk import core
```
Comment (tom98784) [inline python/my-widget-service/my_widget_service/my_widget_service_stack.py]: Can move inside the import statement below

File: python/my-widget-service/setup.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+with open("README.md") as fp:
+    long_description = fp.read()
+
+
+setuptools.setup(
+    name="my_widget_service",
+    version="0.0.1",
+
+    description="An empty CDK Python app",
```
Comment (tom98784) [inline python/my-widget-service/setup.py]: Needs populating with actual info

File: python/my-widget-service/my_widget_service/my_widget_service_stack.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        rest_api_role: iam.Role = iam.Role(
+            self,
+            "RestAPIRole",
+            assumed_by=iam.ServicePrincipal("apigateway.amazonaws.com"),
+            managed_policies=[iam.ManagedPolicy().from_aws_managed_policy_name("AmazonS3FullAccess")]
+        )
+
+        list_objects_response: apigw.IntegrationResponse = apigw.IntegrationResponse()
+        list_objects_response.update({"statusCode": "200"})
+        list_objects_response.update({"statusCode": "200"})
```
Comment (tom98784) [inline python/my-widget-service/my_widget_service/my_widget_service_stack.py]: Duplicate line

File: python/my-widget-service/README.md (hunk 1 above)
Comment (rhboyd) [reply python/my-widget-service/README.md]: I'll add the info. I would be hesitant to use the fact that READMEs haven't been used in the past to justify not including them just to be consistent. I'm personally biased towards a readme for each directory.

File: python/my-widget-service/my_widget_service/my_widget_service.egg-info/PKG-INFO (hunk 2 above)
Comment (rhboyd) [reply python/my-widget-service/my_widget_service/my_widget_service.egg-info/PKG-INFO]: removed

File: python/my-widget-service/README.md (hunk 1 above)
Comment (garnaat) [reply python/my-widget-service/README.md]: The main reason that the existing examples don't have a README is that they were all generated prior to the addition of Python-specific init templates in the CDK CLI tool.  I think the addition of README files is good, although as mentioned, they should be edited to make them relevant to the example.

File: python/my-widget-service/my_widget_service/my_widget_service_stack.py (hunk 9 above)
Comment (rhboyd) [reply python/my-widget-service/my_widget_service/my_widget_service_stack.py]: removed

**Comment by rhboyd** [comment]
I don't quite understand the build failure. it appears to be npm related and not something I can fix in this commit, is this correct?

**Comment by garnaat** [comment]
Yeah, I don't think it's related to your changes and I don't think there is anything you can do to fix it in this PR.

**Comment by rhboyd** [comment]
Is there a way to re-run the tests without pushing another commit? The build was broken when I submitted the change, but the build has been fixed and the test hasn't re-run.

**Comment by garnaat** [comment]
I'm getting merge conflicts when I try to merge locally.  Could we rebase this and see where we stand?

**Comment by rhboyd** [comment]
Yes. Absolutely. I’ll rebase it today and check on the None.None error

**Comment by rhboyd** [comment]
@garnaat this example looks completely broken. I'm okay re-writing it for CDK 1.8 and submitting a fresh PR.

**Pull request closed without merging**